doc/api/url.md

```diff
@@ -880,6 +880,32 @@ console.log(url.domainToUnicode('xn--iñvalid.com'));
 // Prints an empty string
 ```
 
+### url.fileURLToPath(url)
+
+* `url` {URL | string} The file URL string or URL object to convert to a path.
+* Returns: {URL} The fully-resolved platform-specific Node.js file path.
+
+This function ensures the correct decodings of percent-encoded characters as
+well as ensuring a cross-platform valid absolute path string.
+
+When converting from URL to path, the following common errors can occur:
+
+```js
+// '/C:/path/' instead of 'C:\path\' (Windows)
+new URL('file:///C:/path/').pathname;
+
+// '/foo.txt' instead of '\\nas\foo.txt' (Windows)
+new URL('file://nas/foo.txt').pathname;
+
+// '/%E4%BD%A0%E5%A5%BD.txt' instead of '/你好.txt' (POSIX)
+new URL('file:///你好.txt').pathname;
+
+// '/hello%20world.txt' instead of '/hello world.txt' (POSIX)
+new URL('file:///hello world.txt').pathname;
+```
+
+where using `url.fileURLToPath(fileURL)` can get the correct results above.
```

Comment:
Can you please add an example that uses the new API instead?

Reply:
Great point